src/types/rest-types.ts

```diff
@@ -1,8 +1,32 @@
+import { ProjectInvite } from '../types';
+
+export interface Account {
+  id: string;
+  name: string;
+  type: string;
+  description: string;
+  logo: string;
+  products: string[];
+  features: string[];
+  trialActive: boolean;
+  trialEndDate: string;
+}
+
 /**
  * This type definition describes known endpoints in our REST API.
  *
  * It is structured in the format specified by [one-query](https://github.com/lifeomic/one-query).
  */
 export type RestAPIEndpoints = {
-  // TODO: see follow-up PR.
+  'GET /v1/accounts': {
+    Request: {};
```

Comment:
So if we wanted to require the account header, could we add it here as a required part of the request?

Reply:
@sternetj Great question. short answer: no -- `one-query` intentionally doesn't really deal in headers. So, this PR changes nothing about how we'd specify the LifeOmic-Account header for requests -- it can still be provided via `one-query` using the `axios` option, like so:

```typescript
useRestQuery(
  'GET ...',
   { ... },
   {
     axios: { // <-- a special one-query option available adjacent to the standard react-query options
       headers: { ... } // <-- provide header here.
     }
   }
)
```

Longer answer: the "idea" behind one-query is mainly about encouraging a consistent pattern of API interaction. It does this by forcing us to provide a single Axios client that can make all the HTTP requests. In use cases up until now, one-query has only been used in scenarios where we should always be providing the header (rather than only in certain endpoints), so we've always built it in the client directly.

Looking at the SDK code + the various API calls it makes, it seems like it is _much_ more common to _use_ the LifeOmic-Account header than to _not_ use it -- there are only a couple of endpoints that don't need it. It seems this has resulted in code like this being duplicated many times:

```typescript
export function useMyQuery() {
  const { accountHeaders } = useActiveAccount(); // <-- get the account header
  const { httpClient } = useHttpClient();

  return useQuery(
    ['...'],
    () =>
      httpClient
        .get(`/v1/my-url`, { headers: accountHeaders }) // <-- specify it here
        .then((res) => res.data),
    {
      enabled: !!accountHeaders, // <-- don't forget to disable the query in case there is no header
    },
  );
}
```

To me, it seems like it'd be worth optimizing for the common case (using the header) by:
- "building in" this account header logic to the core HTTP client, so that we don't have to so regularly duplicate this code
AND
- providing an escape hatch to allow _not_ specifying the header in scenarios where it shouldn't be provided

I've spent a bit of time trying to make this^ work, and have struggled just a bit. I'll do some more thinking on it. Regardless, I'd like to treat it as "outside the scope of this PR", if that's alright with you.

Reply:
Absolutely! Definitely don't want to hold up the PR for this.
Yeah that pattern is annoying but understandable and as you said it would be nice to switch to an escape hatch model.